meerk40t/core/elements.py

```diff
@@ -5140,94 +5140,115 @@ def classify(self, elements, operations=None, add_op_function=None):
+             2. For each group of raster objects, determine which operations are of the same colour as at least one element in the group.
+                All the raster elements of the group will be added to those operations.
+             3. If there are any raster elements that are not classified in this way then:
+                 A) If there are Default Raster Operation(s), then the remaining raster elements are allocated to those.
+                 B) Otherwise, if there are any non-default raster operations that are empty and those raster operations are of the same colour,
+                    then the remaining raster operations will be allocated to those Raster Operations.
+                 C) Otherwise, a new Default Raster operation will be created and remaining Raster elements will be added to that.
+
+        The current code does not do the following:
+        a)  Handle rasters in second or later files which overlap elements from earlier files which have already been classified into operations.
+            It is assumed that if they happen to overlap that is coincidence. After all the files could have been added in a different order and
+            then would have a different result.
+        b)  Handle any differently the reclassifications of single elements which have e.g. had their colour changed. (This needs to be checked.)
 
         :param elements: list of elements to classify.
         :param operations: operations list to classify into.
         :param add_op_function: function to add a new operation, because of a lack of classification options.
         :return:
         """
-        reverse = self.context.classify_reverse
-        # If reverse then we insert all elements into operations at the beginning rather than appending at the end
-        # EXCEPT for Rasters which have to be in the correct sequence.
-        element_pos = 0 if reverse else None
+
         if elements is None:
             return
         if operations is None:
             operations = list(self.ops())
         if add_op_function is None:
             add_op_function = self.add_classify_op
 
+        reverse = self.context.classify_reverse
+        # If reverse then we insert all elements into operations at the beginning rather than appending at the end
+        # EXCEPT for Rasters which have to be in the correct sequence.
+        element_pos = 0 if reverse else None
+
+        vector_ops = []
+        raster_ops = []
+        special_ops = []
         default_cut_ops = []
         default_engrave_ops = []
         default_raster_ops = []
+        rasters_one_pass = None
 
-        raster_ops = []
-        deferred_raster_elements = []
-        white_raster_elements = []
-        have_non_white_deferred_rasters = False
-
-        raster_ops_single_color = None
         for op in operations:
+            if op.operation in ["Cut", "Engrave"]:
```

Comment:
Making this a list() is unneeded when a tuple() will do

Reply:
Loads of other lists that didn't need to be mutable in the raster grouping code. So I fixed those too.